src/lib/connect.ts

```diff
@@ -204,12 +210,23 @@ export const connect = (
   const typedRpcSubscriptions = rpcSubscriptions as ReturnType<typeof createSolanaRpcSubscriptions>;
 
   // Create the transaction confirmation functions based on the cluster name
-  const sendAndConfirmTransaction = clusterNameOrURL.includes("mainnet")
-    ? sendAndConfirmTransactionFactory({ rpc, rpcSubscriptions: typedRpcSubscriptions as any })
-    : clusterNameOrURL.includes("testnet")
-      ? sendAndConfirmTransactionFactory({ rpc, rpcSubscriptions: typedRpcSubscriptions as any })
-      : sendAndConfirmTransactionFactory({ rpc, rpcSubscriptions: typedRpcSubscriptions as any });
-
+  let sendAndConfirmTransaction: ReturnType<typeof sendAndConfirmTransactionFactory>;
+  
+  switch (true) {
+    case clusterNameOrURL.includes("mainnet"):
+      sendAndConfirmTransaction = sendAndConfirmTransactionFactory({ rpc, rpcSubscriptions: typedRpcSubscriptions as RpcSubscriptionsMainnet<SolanaRpcSubscriptionsApi> });
```

Comment:
Thanks I've been meaning to fix this for a while. 😃 Would:


```
rpcSubscriptions: RpcSubscriptionsMainnet<SolanaRpcSubscriptionsApi>
```

work? I don't understand why we need the type coersion.